azure-tests/src/main/java/fixtures/paging/Pagings.java

```diff
@@ -202,73 +202,109 @@
         @ExpectedResponses({200})
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<ProductResultValue>> getNoItemNamePagesNext(
-                @PathParam(value = "nextLink", encoded = true) String nextLink, Context context);
+                @PathParam(value = "nextLink", encoded = true) String nextLink,
+                @HostParam("$host") String host,
```

Comment:
Will azure-core work correctly, when `host` and `nextLink` both contains host part of the URL?

Reply:
@weidongxu-microsoft  yes, this is how the previous version of autorest also generated the code which worked with azure core. So, this should work.

Reply:
I didn't get it. Previous version of autorest generate this
```
        Mono<Response<ResourceGroupListResult>> listNext(
            @PathParam(value = "nextLink", encoded = true) String nextLink, Context context);
```
while updated code had one more `@HostParam("$host") String host`, which certainly is not same?

Well, if tests passes, I assume azure-core could handle this.

Reply:
nit. Will it be better to put `@HostParam` before `@PathParam`? Not important since this method will not be exposed to user.

Reply:
Sorry, I meant the v3 version of autorest. Here's an example of what was generated by v3.

```java
    @Get("{nextUrl}")
    @ExpectedResponses({200})
    @UnexpectedResponseExceptionType(HttpResponseException.class)
    @ReturnValueWireType(ConfigurationSettingPage.class)
    Mono<PagedResponse<ConfigurationSetting>> listKeyValues(
        @HostParam("url") String url,
        @PathParam(value = "nextUrl", encoded = true) String nextUrl,
        Context context);
```